public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -162,6 +162,11 @@ export class LokiDatasource
     this.logContextProvider = new LogContextProvider(this);
   }
 
+  /**
+   * This method is implemented for DataSourceWithSupplementaryQueriesSupport and it retrieves a data provider
```

Comment:
```suggestion
   * Implemented for DataSourceWithSupplementaryQueriesSupport. It retrieves a data provider
```

I would consider removing all the "This method", because it seems redundant in the context of code comments/JS doc. Might be a nitpick, but I think it makes a difference. E.g:

```
// This variable holds the biggest possible integer for X and Y
const max = Number.MAX_SAFE_INTEGER;

// vs

// Biggest possible integer for X and Y
const max = Number.MAX_SAFE_INTEGER;
```

Reply:
Yeah, I totally agree. Fixed.